vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/helpers/ResponseModeParametersFactory.kt

```diff
@@ -0,0 +1,93 @@
+package at.asitplus.wallet.lib.oidc.helpers
+
+import at.asitplus.KmmResult
+import at.asitplus.wallet.lib.oidc.AuthenticationRequestParameters
+import at.asitplus.wallet.lib.oidc.OpenIdConstants
+import at.asitplus.wallet.lib.oidvci.OAuth2Exception
+import io.github.aakira.napier.Napier
+
+
+class ResponseModeParametersFactory(val requestParameters: AuthenticationRequestParameters) {
+    fun createResponseModeParameters(): KmmResult<ResponseModeParameters> =
+        when (requestParameters.responseMode) {
+            OpenIdConstants.ResponseMode.DIRECT_POST -> createDirectPostResponseModeParameters()
+            OpenIdConstants.ResponseMode.DIRECT_POST_JWT -> createDirectPostJwtResponseModeParameters()
+            OpenIdConstants.ResponseMode.QUERY -> createQueryResponseModeParameters()
+
+            // default for vp_token and id_token is fragment
+            null, OpenIdConstants.ResponseMode.FRAGMENT -> createFragmentResponseModeParameters()
+            else -> TODO()
+        }.getOrElse {
+            return KmmResult.failure(it)
```

Comment:
same as above: why not using `KmmResult.wrap()` extension function?

Reply:
fixed in commit 9eb05adebb149f067312e3d8bbfec2c56a0cbd60